File: Module_work/animal.py

# someVariable = 3

# def somefunction():
#     print("Hello")
#%%
class Animal():

    def __init__(self,animalName,animalType,animalSound) -> None:
        self.animalName = animalName
        self.animalType = animalType
        self.animalSound = animalSound
    
    def printName(self):
        print(self.animalName)
#%%
motelyObject = Animal(animalName='Motely',animalType='Cat',animalSound='Meow')
# %%
motelyObject.animalName
# %%
motelyObject.printName()
# %%
laikaObject = Animal(animalName='Laika',animalType='Dog',animalSound='Woof')
# %%
laikaObject.animalName
# %%
laikaObject.printName()
# %%
laikaObject.canFetch = True
#%%
laikaObject.someNewMethod = def printCanFetch(self):
    print(self.canFetch)
#%%
laikaObject.canFetch
# %%

File: Module_work/animal.js
function someFunction(){
    console.log("hello!")
}

function Animal(animalName, animalType, animalSound){
    this.animalName = animalName; // 'this' is the equivalent to 'self' in python
    this.animalType = animalType;
    this.animalSound = animalSound;


}

motelPrototype = Animal('Motely', 'Cat', 'Meow' );

function printSound(someObject){
    console.log(someObject.animalSound);
}

motelPrototype.printSound();

laikaPrototype = Animal('laika', 'dog', 'woof')

